exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -508,7 +511,10 @@ public void populatePruningVector(ValueVector v, int index, SchemaPath column, S
         NullableVarBinaryVector varBinaryVector = (NullableVarBinaryVector) v;
         Object s = partitionValueMap.get(f).get(column);
         byte[] bytes;
-        if (s instanceof Binary) {
+        if (s == null) {
+          varBinaryVector.getMutator().setNull(index);
```

Comment:
We can apply this setNull() only for types that use BinaryStatistics [1] for the reason described in my previous comment. So we apply setNull() to VARBINARY, DECIMAL18, VARCHAR and INTERVAL types.

[1] https://github.com/apache/parquet-mr/blob/e54ca615f213f5db6d34d9163c97eec98920d7a7/parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java#L25